### ISSUE TITLE
[macOS] Yuzu build failure after #10082

### ISSUE THREAD

**Issue by shinra-electric** [yuzu-emu/yuzu#10136]
### Issue

Yuzu no longer builds after #10082 

Checking out the previous commit `0f1ff5f`  using git and Yuzu builds fine. #10076
[Terminal Output 0f1ff5f (success).txt](https://github.com/yuzu-emu/yuzu/files/11374860/Terminal.Output.0f1ff5f.success.txt)

Checking out `fe57f39` and the build fails. #10082 
[Terminal Output fe57f39 (fail).txt](https://github.com/yuzu-emu/yuzu/files/11374862/Terminal.Output.fe57f39.fail.txt)

System: 
MBP M1 Pro
Ventura 13.3.1
Xcode 14.3

**Comment by Kelebek1** [comment]
https://github.com/yuzu-emu/yuzu/blob/master/src/video_core/texture_cache/texture_cache.h#L891-L894

Maybe an issue with that array being given a size_t and a VkDeviceSize? They should alias, but maybe your compiler hates it. Try casting download_map.offset to a size_t.

**Comment by shinra-electric** [comment]
Just a quick disclaimer, I'm rubbish at C++. 

I edited
```
        std::array buffer_offsets{
            buffer_offset,
            download_map.offset,
        };
```

to: 
```
        std::array buffer_offsets{
            buffer_offset,
            static_cast<size_t>(download_map.offset),
        };
```

Is this what you meant? 
(If so, it did not fix it).

**Comment by shinra-electric** [comment]
This is fixed by #10143
